docs/application/concepts/170-records.mdx

```diff
@@ -0,0 +1,101 @@
+---
+title: Records
+description:
+  Immutable data classes in Java that are heavily used in the Telestion
+  Ecosystem
+---
+
+<!--
+Concept
+=======
+
+=== When to use this template:
+Use this template when you need to describe a concept, technology,
+or something similar. In other words: whenever you need to write an
+article that describes a concept where the concept's name can be the
+article's title (e.g., "Vert. x", "Verticle", "Event Bus")
+
+=== When not to use this template:
+Do not use these templates for articles that give context on how
+to use/approach the topic. Instead, it would be best if you used a guide for that.
+
+Writing a guide and a concept article are not mutually exclusive. Instead,
+you should, in most cases, write both, where the concept gives a brief overview
+of what it is, and the guide is a longer article explaining best practices on using
+it.
+
+E.g.:
+Concept: Event Bus
+Guide: Interacting with the Event Bus
+
+
+=== Writing tips:
+- Write in the present tense
+- Use neutral pronouns (they/them instead of he/she and him/her)
+- Be respectful to everyone
+- Be aware of the potential for cultural misunderstandings
+-->
+
+<!-- Relevant imports: -->
+
+import { Reference, Image } from '/components';
+
+<!-- Brief Description (1-3 lines) of the concept: -->
+
+Records are immutable data classes used in Java to encapsulate "packets" of
+data.
+
+Records make it much easier to implement these kinds of "wrapper classes"
+without having to define constructors, `toString()`, etc., every single time.
```

Comment:
Imo getters and setters are even more important.

Reply:
RE getters: agreed, I'm on it